src/main/scala/is/hail/expr/types/MatrixType.scala

```diff
@@ -184,4 +140,19 @@ case class MatrixType(
     newline()
     sb += '}'
   }
+
+  def copyParts(globalType: TStruct = globalType,
+    colKey: IndexedSeq[String] = colKey,
+    colType: TStruct = colType,
+    rowPartitionKey: IndexedSeq[String] = rowPartitionKey,
+    rowKey: IndexedSeq[String] = rowKey,
+    rowType: TStruct = rowType,
+    entryType: TStruct = entryType): MatrixType = {
+    if (this.rowPartitionKey.nonEmpty && rowPartitionKey.isEmpty) {
+      warn(s"deleted row partition key [${this.rowPartitionKey.mkString(", ")}], " +
+        s"all downstream operations will be single-threaded")
```

Comment:
This doesn't warn in the case when you create a un-keyed MT ab initio, for example, when loading a TSV.  Should we warn in fromParts but have it take a alreadyWarned: Boolean = false and pass in alreadyWarned = this.rowPartitionKey.isEmpty here?

Reply:
how can you create an unkeyed MT from a TSV?

Reply:
Oh,  hmm.  I was thinking import_matrix, but I guess the interface current requires a key?  This is going to need some rethinking going forward (e.g. unpartitioned MT and import_matrix shouldn't have a key expr as per our recent conversations and we should warn only when doing partitioned operations when the keys are empty.)

Reply:
yeah, exactly. only warn before a join/shuffle

Reply:
in the current model I think this is safe -- all matrices are created with a key, so you will get a warning at least once if you remove your keys